pkg/engine/task/task_apply.go

```diff
@@ -239,16 +241,6 @@ func isHealthy(ro []runtime.Object) error {
 	return nil
 }
 
-func isTerminallyFailed(ro []runtime.Object) error {
```

Comment:
Arguably this is a better separation of concerns: determining whether a resource health error is terminal is better left to the `HealthUtil`